proposals/interface-types/working-notes/choices/webidl.md

```diff
@@ -0,0 +1,270 @@
+interoperability as it applies to WebAssembly is the role of the function
+call. Unlike interoperability expressed at the level of web services (say),
+using a foreign capability in WebAssembly is most naturally expressed as a
+function call.
+
+I.e., invoking capabilities does not naturally imply sending messages,
+serializing data, network connections, or even crossing a process boundary: the
+invoked capability may even be present in the same thread.
+
+This context informs much of the style of how interoperability is expressed in
+the Interface Types proposal. In particular, this proposal does not include a
+serialization format.[^Although there is a binary encoding, its role is to
+support the publishing and loading of WebAssembly modules; not to support
+function interaction.] Indeed, it is expected that no serialization is required
+when invoking a function via Interface Types' adapters.
+
+### Language Specific Module Systems
+
+Interface Types are _not_ intended to replacement language specific module
+systems. Nor are Interface Types intended to address the general inter-language
+interoperability problem.
+
+Indivual languages will often have their own techniques aimed at allowing
+importing and exporting of modules. For example, C/C++ modules would continue to
+be linked together into single WebAssembly modules.
+
+## WebIDL
+
+[Web IDL][WebIDL] is clearly an important IDL for Web APIs. And, since the web
+platform is a crucial domain for WebAssembly, it seems necessary to address the
+question of "why not WebIDL" for describing APIs in Interface Types.
+
+The case against WebIDL is comprised of three principal arguments:
+
+* WebIDL is too strongly aligned to JavaScript;
+* WebIDL has features that are not relevant in the context of WebAssembly; and
+* WebIDL does not support certain necessary features.
+
+### JavaScript
+
+WebAssembly can be characterized as a vehicle that allows languages _other than_
+JavaScript to be executed on the web (specifically, in a browser tab).
+
+On the other hand, WebIDL is designed to be an IDL for JavaScript. This results
+in WebIDL having features that may be problematic for other languages.
+
+For example, WebIDL has the concepts of `object` and _Indexed Properties_; where
+`object` means the value is a JavaScript object and an _Indexed Property_ is one
+which exposes an interface that allow indexed access.
+
+Many languages do not support these concepts; and it may be quite onerous to do
+so. However, any feature of an IDL should be useable by any API designer to
+create an appropriate API. Features of an IDL that target JavaScript should not
+be used by an API designer that wishes to design for many different languages.
+
+There is a fine distinction to be made in making 'avoidance' recommendations and
+constructing a completely new IDL.
+
+#### WebIDL Strings
+
+Certain aspects of WebIDLs design seem to be somewhat overelaborated. For
```

Comment:
I'd avoid saying "overelaborated" b/c it sounds judgmental.  Really, the points in this subsection support the claim in the preceding section (WebIDL is over-specialized on JS), so I'd just delete this sentence and the "WebIDL Strings" heading, fusing the following two paragraphs with the preceding subsection.

Reply:
I do believe that we need to call out the different flavors of strings and floats. It is unique among IDLs.

Reply:
If you look closely, there is an extra heading level here. It is part of the Javascript story.